src/lib/logger.ts

```diff
@@ -0,0 +1,61 @@
+type severityLevels = -1 | 0 | 1 | 2 | 3 | 4 | 5; // (-1 = meta, 0 = info, 1 = debug, 2 = warn, 3 = error, 4 = fatal, 5 = shutdown)
+type outputObject = {
+	level: severityLevels;
+	text: string;
+};
+
+const output: outputObject[] = [];
+
+const backslash = string.char(92); // for some reason I can't store a backslash as a single character. ._.
+
+class Scope {
+	name = "unnamed";
+
+	constructor(scopeName: string, printToFile: boolean) {
+		this.name = scopeName;
+	}
+
+	static scope(scopeName: string, printToFile: boolean) {
+		return new this(scopeName, printToFile);
+	}
+
+	static getOutput() {
+		return output;
+	}
+
+	log(severity: severityLevels, ...toPrint: any[]) {
+		let stringToPrint = "";
+		toPrint.forEach((value) => {
+			stringToPrint = `${stringToPrint}${tostring(value)} `;
+		});
+
+		output.unshift({
+			level: severity,
+			text: `[${this.name}]: ${stringToPrint}`,
+		});
```

Comment:
Please don't do this, very bad performance

Reply:
I'm confused. Are you talking about the concat, or the unshift?

String interpolation is not performance heavy, and .unshift() is a single table.insert call, which isn't bad performance.

Reply:
It's the .unshift, just make it append to end because that's how we're going to be displaying anyway.